app.rb

```
@@ -0,0 +1,30 @@
+  missing_keys = expected_keys - test_results.keys
+  extra_keys = test_results.keys - expected_keys
+
+  unless missing_keys.empty? && extra_keys.empty?
+    halt 400, { "error": "Invalid request payload", "missing_keys": missing_keys, "extra_keys": extra_keys }.to_json
+  end
+
+  # Process results and return with 201 status
+  pass_fail_results = TestResultsProcessor.new(test_results)
+  halt 201, pass_fail_results.add_pass_rate.to_json
+
+end
```

rubocop

[convention] Extra empty line detected at block body end.

(Layout/EmptyLinesAroundBlockBody)